### ISSUE TITLE
Prices API reports wildly different numbers from call to call

### ISSUE THREAD

**Issue by mtlynch** [NebulousLabs/Sia#2364]
I've noticed that the `siac renter prices` output varies wildly from call to call. For example, I just ran the command:

```
$ $GOPATH/bin/siac renter prices
Renter Prices (estimated):
  Fees for Creating a Set of Contracts:   103.5 SC
  Download 1 TB:                          45.24 SC
  Store 1 TB for 1 Month:                 258.5 SC
  Upload 1 TB:                            57.13 SC
```

Then ran it again a second later:

```
$ $GOPATH/bin/siac renter prices
Renter Prices (estimated):
  Fees for Creating a Set of Contracts:   101 SC
  Download 1 TB:                          34.42 SC
  Store 1 TB for 1 Month:                 241.7 SC
  Upload 1 TB:                            20.65 SC
```

* **Expected behavior**: `siac renter prices` reports consistent prices when called a few seconds apart.
* **Actual behavior**: `siac renter prices` reports prices that are wildly different between calls, in some cases up to 276% higher.

I'm guessing this is due to the fact that there's randomness in picking host contracts. Would it be possible for this to be random, but repeatable so that given the same set of hosts with the same offerings, the client chooses the same random set of contracts?

**Comment by tbenz9** [comment]
@mtlynch I'm not sure exactly what the purpose of this Issue is.  If you're asking a question I would recommend asking it on Discord.  If you're making a feature-request I would recommend coming up with some kind of suggested solution (maybe even submitting a Pull Request).

**Comment by mtlynch** [comment]
@tbenz9, thanks for triaging these.

> I'm not sure exactly what the purpose of this Issue is. If you're asking a question I would recommend asking it on Discord. If you're making a feature-request I would recommend coming up with some kind of suggested solution (maybe even submitting a Pull Request).

This is not a question or a feature request. It is a bug report. The behavior I reported is unexpected and degrades user experience, so it's a bug.

>Just to clarify, I agree that the renter prices are a little strange. But i'm not sure what you hope to accomplish with this Issue, and since no one has commented in 11 days I'd guess they are confused as well.

I don't think that means much. There are 162 open bugs and until today, nobody's been triaging them, so lots of them have no comment from the dev team.

**Comment by tbenz9** [comment]
Thanks for the clarification.  My understanding is that since Sia is a storage marketplace the sellers can set their own prices which is one reason the renter prices have such a large range.  As to why they fluctuate every time you run `renter prices` is because Sia does not currently support dynamic host pricing.  This feature is on the roadmap and should smooth out the variability when you run `renter prices`.  Until then, this is certainly undesirable behavior, but not necessarily a bug, it's reporting the prices based on the sample of hosts it polled at that moment, even if they change with no real repeatability, that's how it was designed (as of now).

**Comment by mtlynch** [comment]
>As to why they fluctuate every time you run renter prices is because Sia does not currently support dynamic host pricing. This feature is on the roadmap and should smooth out the variability when you run renter prices.

I don't think this is correct. Dynamic host pricing will make the prices change much *more* in SC amount, because it will let hosts peg their prices to a fiat currency rather than a fixed amount in SC. But even then, market price is not going to change by a factor of 3x within a few seconds.

>Until then, this is certainly undesirable behavior, but not necessarily a bug,

Sorry, what's the distinction  in this case between undesirable program behavior and a bug?

>it's reporting the prices based on the sample of hosts it polled at that moment, even if they change with no real repeatability, that's how it was designed (as of now).

The hosts are not changing their prices every few seconds. I suspect Sia is using a random subset of hosts to report pricing on every call. I think it should be possible to make this subset random across users but repeatable for the same user and the same set of host advertisements (e.g. seeded based on a hash of the user's most recently generated wallet address).

**Comment by avahowell** [comment]
I think this definitely qualifies as a bug. The solution @mtlynch proposed seems reasonable to me. A user definitely expects that repeated calls to `prices` remain relatively stable, it's very confusing and worrisome if the reported prices jump around by this much.

**Comment by tbenz9** [comment]
> I think it should be possible to make this subset random across users but repeatable for the same user and the same set of host advertisements

I think that's a great suggestion.

Recommendation to Sia team: tag with 'Programming Task'.

**Comment by tbenz9** [comment]
I think this was fixed with #2434 but not closed.

Recommendation to devs: close
